ledger/src/shred/wire.rs

```diff
@@ -163,6 +172,32 @@ pub(crate) fn get_data(shred: &[u8]) -> Result<&[u8], Error> {
     }
 }
 
+/// Returns the ErasureConfig specified by the coding shred
+/// Caller must verify that the shred is a coding shred
+#[inline]
+pub(crate) fn get_erasure_config(shred: &[u8]) -> Result<ErasureConfig, Error> {
+    debug_assert_eq!(get_shred_type(shred).unwrap(), ShredType::Code);
+    let Some(num_data_bytes) = shred.get(83..83 + 2) else {
```

Comment:
would it not be nicer to return Error of some sort if shred type is wrong?

Reply:
yeah that's fine, we'll remove this in the next shred version anyway [c15f72f](https://github.com/anza-xyz/agave/pull/7728/commits/c15f72f6e5139b2b1c7728ca486c157ed8e1121a)